server/game/cards/challengeevent.js

```diff
@@ -0,0 +1,48 @@
+const DrawCard = require('../drawcard.js');
+
+class ChallengeEvent extends DrawCard {
+    // event card that can only be played during a challenge
+
+    constructor(owner, cardData, challengeType, partecipatingAs) {
+        // - challengeType (optional): make the event playable only on a
+        //   challenge of the given type
+        //
+        // - partecipatingAs (optional): make the event playable only when the
+        //   event controller is partecipating as either 'attacker' or
+        //   'defender'
+        //
+        super(owner, cardData);
+
+        this._challengeType = challengeType;
+        this._partecipatingAs = partecipatingAs;
```

Comment:
Spelling - `participatingAs`. Also unsure of the convention of prefixing these variables with underscore. I understand what you're going for here, but it's not something we've done elsewhere.